### DISCUSSION TITLE
Integrating Crate for Javascript query search

### DISCUSSION THREAD

**Discussion by ghost** [crate/crate#16020] [General]
Hi All, 
I am building a new web application and want to hear from you all that is there a plugin or support for the installation of the CrateDB locally on a server side and let javascript or angular interact with it the same. Thank you for the answers. if there are any previous links or style guide documentation for the same then kindly let me know. 

Alles gute,
Gaurav

**Comment by hlcianfagna** [comment]
Hi,

There are multiple ways to install CrateDB locally on your servers, see https://cratedb.com/docs/guide/install/

and support is available via support plans or through the Community see https://cratedb.com/support

Regarding the interaction of your web application with CrateDB, 
- you can either directly use the [HTTP endpoint](https://cratedb.com/docs/crate/reference/en/latest/interfaces/http.html)
- or one of the many drivers available for different programming languages https://cratedb.com/docs/crate/clients-tools/en/latest/connect/index.html

You may also be interested in the fully-managed DBaaS option https://cratedb.com/docs/cloud/en/latest/index.html

**Comment by ghost** [comment]
Hi, 

Thank you for the reply. 

Gaurav

**Comment by ghost** [comment]
Hi, 
I installed using the guideline listed but it is showing connect error. I installed the crash to built an in-feed system and write a connector but it is not connecting. Any support for the installation. 

Thank you,
Gaurav

**Comment by hlcianfagna** [comment]
Hi,
As a starting point you may want to add `--verbose` to your `crash` command line to get more details on the connection error, it could be that it cannot reach CrateDB or that there is some authentication issue.
In general you want to confirm CrateDB is running (try to connect locally - try the admin UI - check the logs) and that there is network connectivity (name resolution - routing - ports - etc) between your client machine and the CrateDB machine you are trying to reach.

**Comment by ghost** [comment]
Hi, 
Thank you and I will do the same. Actually, i am writing a javascript connector to connect the same to the online application so that it fetches the submitted queries and dynamically updated the results. I will see this and will let know if i face any connection problems. 

Thank you for the reply,
Gaurav

**Comment by robd003** [comment]
You might find this helpful: https://www.npmjs.com/package/node-crate

**Comment by ghost** [comment]
Thank you and i was reading this only yesterday night and also how to integrate the RxJs with this. Thank you for pointing to this resource which might help also others.
